BlockchainSdk/Blockchains/TON/TONWalletManager.swift

```diff
@@ -56,20 +56,43 @@ final class TONWalletManager: BaseManager, WalletManager {
     ) -> AnyPublisher<TransactionSendResult, SendTxError> {
         getJettonWalletAddressIfNeeded(for: transaction.sourceAddress, transactionType: transaction.amount.type)
             .receive(on: DispatchQueue.global())
-            .tryMap { [weak self] jettonWalletAddress -> String in
+            .tryMap { [weak self] jettonWalletAddress -> (TONTransactionInput, Data) in
                 guard let self else {
                     throw WalletError.failedToBuildTx
                 }
 
                 let params = transaction.params as? TONTransactionParams
 
-                let input = try txBuilder.buildForSign(
+                let buildInput = TONTransactionInput(
                     amount: transaction.amount,
                     destination: transaction.destinationAddress,
+                    expireAt: createExpirationTimestampSecs(),
                     jettonWalletAddress: jettonWalletAddress,
                     params: params
                 )
-                return try buildTransaction(input: input, with: signer)
+
+                let hashForSign = try transactionBuilder.buildForSign(buildInput: buildInput)
+
+                return (buildInput, hashForSign.data)
+            }
+            .withWeakCaptureOf(self)
+            .flatMap { manager, input in
+                let (buildInput, hash) = input
+                let transactionInputPublisher = Just(buildInput).setFailureType(to: Error.self)
+                let signaturePublisher = signer.sign(hash: hash, walletPublicKey: manager.wallet.publicKey)
+
```

Comment:
nit: Можно и без Just() я думаю)

```suggestion
                return signer.sign(hash: hash, walletPublicKey: manager.wallet.publicKey)
                .map { ($0, buildInput) }
```